scene/gui/tree.cpp

```diff
@@ -885,19 +1058,38 @@ void TreeItem::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_button_disabled", "column", "button_idx", "disabled"), &TreeItem::set_button_disabled);
 	ClassDB::bind_method(D_METHOD("is_button_disabled", "column", "button_idx"), &TreeItem::is_button_disabled);
 
-	ClassDB::bind_method(D_METHOD("set_expand_right", "column", "enable"), &TreeItem::set_expand_right);
-	ClassDB::bind_method(D_METHOD("get_expand_right", "column"), &TreeItem::get_expand_right);
-
 	ClassDB::bind_method(D_METHOD("set_tooltip", "column", "tooltip"), &TreeItem::set_tooltip);
 	ClassDB::bind_method(D_METHOD("get_tooltip", "column"), &TreeItem::get_tooltip);
 	ClassDB::bind_method(D_METHOD("set_text_align", "column", "text_align"), &TreeItem::set_text_align);
 	ClassDB::bind_method(D_METHOD("get_text_align", "column"), &TreeItem::get_text_align);
-	ClassDB::bind_method(D_METHOD("move_to_top"), &TreeItem::move_to_top);
-	ClassDB::bind_method(D_METHOD("move_to_bottom"), &TreeItem::move_to_bottom);
+
+	ClassDB::bind_method(D_METHOD("set_expand_right", "column", "enable"), &TreeItem::set_expand_right);
+	ClassDB::bind_method(D_METHOD("get_expand_right", "column"), &TreeItem::get_expand_right);
 
 	ClassDB::bind_method(D_METHOD("set_disable_folding", "disable"), &TreeItem::set_disable_folding);
 	ClassDB::bind_method(D_METHOD("is_folding_disabled"), &TreeItem::is_folding_disabled);
 
+	ClassDB::bind_method(D_METHOD("create_child", "idx"), &TreeItem::create_child, DEFVAL(-1));
+	ClassDB::bind_method(D_METHOD("get_tree"), &TreeItem::get_tree);
+
+	ClassDB::bind_method(D_METHOD("get_next"), &TreeItem::get_next);
+	ClassDB::bind_method(D_METHOD("get_prev"), &TreeItem::get_prev);
+	ClassDB::bind_method(D_METHOD("get_parent"), &TreeItem::get_parent);
+	ClassDB::bind_method(D_METHOD("get_first_child"), &TreeItem::get_first_child);
+
+	ClassDB::bind_method(D_METHOD("get_next_visible", "wrap"), &TreeItem::get_next_visible, DEFVAL(false));
+	ClassDB::bind_method(D_METHOD("get_prev_visible", "wrap"), &TreeItem::get_prev_visible, DEFVAL(false));
+
+	ClassDB::bind_method(D_METHOD("get_sibling", "nth"), &TreeItem::get_sibling);
+	ClassDB::bind_method(D_METHOD("get_first_sibling"), &TreeItem::get_first_sibling);
```

Comment:
In general in the API we don't have so many function to get the same things. I think get_sibling and get_sibling_count, are enough, that's how most APIs are.

Reply:
Well, to be honest I don't really understand, this is quite an unusual API. 

If I understand the purpose of this API, we simply want it to be able to direct children of a TreeItem, am I right?

If that's so I believe should make something simpler, like get_direct_child(i) and get_direct_children_count(). Possibly we can add a get_index_as_direct_child() or something like that, to check if we not getting the current TreeItem.

Reply:
After thinking a bit, I came up to the conclusion that get_direct_child(idx) will be the same of get_first_sibling().get_sibling(idx), so there is no real gain. 
It really depends on if you prefer to always get items starting from the first (from the parent) and use indexes, 
or you prefer to get items in a relative way and keep the absolute solution as an edge case.

At this point for me is the same, but I still think that my solution is the most flexible. 
I hope I clarify it.

Reply:
And, I'm going to remove get_last_sibling. I make it to pair it with get_first_sibling, but it's really an edge case compared to get_first_sibling.

Reply:
Well, I think that for consistency with the remaining of the API, getting the children from the parent make more sense. So I would go that way, yes. It's kind of easier to understand.

Reply:
Ok, I remove all the get_sibling_* methods and replaced it with a Node-style API (get_child, get_child_count, get_children, get_index)